### PR TITLE
feat(route): add query keyword parse of cool paper

### PR DESCRIPTION
**Pull request by Muyun99**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/papers/query/Detection
/papers/arxiv/cs.AI
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
增加了 cool paper RSS 源对 query keyword的解析，可以传入一个 keyword，然后自动解析

### CODE REVIEW

File: lib/routes/papers/index.ts [hunk 1]
```diff
@@ -81,19 +92,25 @@ export const route: Route = {
     url: 'papers.cool',
     maintainers: ['nczitzk', 'Muyun99'],
     handler,
-    example: '/papers/arxiv/cs.AI',
-    parameters: { category: 'Category, arXiv Artificial Intelligence (cs.AI) by default' },
+    example: '/papers/arxiv/cs.AI or /papers/query/Detection',
```
Comment (TonyRL) [inline lib/routes/papers/index.ts:95]: This example will be converted into `"https://rsshub.app/papers/arxiv/cs.AI or /papers/query/Detection"`

**Comment by Muyun99** [comment]
@TonyRL Hello, Tony, I have finished my modification, split the arxiv/category and query/keyword. Please help me to review the code  again, thanks

File: lib/routes/papers/query.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    },
+    radar: [
+        {
+            title: 'arXiv Paper queryed by Detection',
+            source: ['papers.cool/arxiv/search?highlight=1&query=Detection&sort=0`'],
+            target: '/papers/query/Detection',
+        },
+        {
+            title: 'arXiv Paper queryed by Segmentation',
+            source: ['papers.cool/arxiv/search?highlight=1&query=Segmentation&sort=0`'],
```
Comment (TonyRL) [inline lib/routes/papers/query.ts:114]: https://github.com/DIYgod/RSSHub/pull/17894#discussion_r1887315936

File: lib/routes/papers/query.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            },
+            language,
+            enclosure_url: pdfUrl,
+            enclosure_type: 'application/pdf',
+            enclosure_title: title,
+        };
+    });
+
+    return {
+        title: feed.title,
```
Comment (TonyRL) [inline lib/routes/papers/query.ts:68]: Feed title is [undefined](https://github.com/DIYgod/RSSHub/pull/17894#issuecomment-2551747210)

File: lib/routes/papers/query.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+const pdfUrlGenerators = {
+    arxiv: (id: string) => `https://arxiv.org/pdf/${id}.pdf`,
+};
+
+export const handler = async (ctx) => {
+    const { keyword = 'query/Detection' } = ctx.req.param();
+    const limit = ctx.req.query('limit') ? Number.parseInt(ctx.req.query('limit'), 10) : 150;
+
+    const rootUrl = 'https://papers.cool';
+    const query = keyword.split(/\//)[1];
```
Comment (TonyRL) [inline lib/routes/papers/query.ts:19]: `query` is [undefined](https://github.com/DIYgod/RSSHub/actions/runs/12396998844/job/34606336199#step:13:20)

Comment (Muyun99) [reply lib/routes/papers/query.ts:19]: 和 lib/routes/papers/index.ts 一致，从 keyword 中解析

File: lib/routes/papers/query.ts (hunk 3 above)
Comment (Muyun99) [reply lib/routes/papers/query.ts:68]: 和 lib/routes/papers/index.ts 一致，feed 是有 title 的

Comment (TonyRL) [reply lib/routes/papers/query.ts:68]: image

Comment (Muyun99) [reply lib/routes/papers/query.ts:68]: 哦哦明白，这个意思，我再去修改一下，感谢

File: lib/routes/papers/query.ts (hunk 4 above)
Comment (Muyun99) [reply lib/routes/papers/query.ts]: 已经完成修复

File: lib/routes/papers/query.ts (hunk 3 above)
Comment (Muyun99) [reply lib/routes/papers/query.ts:67]: 已经完成修复

**Pull request merged**